sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -73,7 +74,7 @@ private[sql] class JacksonGenerator(
 
   private val gen = {
     val generator = new JsonFactory().createGenerator(writer).setRootValueSeparator(null)
-    if (options.pretty) generator.useDefaultPrettyPrinter() else generator
```

Comment:
`useDefaultPrettyPrinter()` should return the same default printer, see the method's doc:
```
    /**
     * Convenience method for enabling pretty-printing using
     * the default pretty printer
     * ({@link com.fasterxml.jackson.core.util.DefaultPrettyPrinter}).
     *
     * @return This generator, to allow call chaining
     */
    public abstract JsonGenerator useDefaultPrettyPrinter();
```
Could you clarify how your code changes the behavior, please.

Reply:
Of course I read the doc but the actually returned instance is not equivalent to the instance created by `new DefaultPrettyPrinter("")`.
https://github.com/FasterXML/jackson-core/blob/jackson-core-2.12.2/src/main/java/com/fasterxml/jackson/core/base/GeneratorBase.java#L452

The parameter of the constructor is the separator which separate multiple root JSON records and the default separator character is `" "`.
https://github.com/FasterXML/jackson-core/blob/jackson-core-2.12.2/src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java#L101
https://github.com/FasterXML/jackson-core/blob/jackson-core-2.12.2/src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java#L28

Reply:
Thank you for the clarification.